gui/proxy_edit.go

```diff
@@ -2,11 +2,55 @@ package gui
 
 import (
 	"github.com/gotk3/gotk3/gtk"
+	"github.com/twstrike/coyim/i18n"
 	"github.com/twstrike/coyim/net"
 )
 
+// this is an array because it must be sorted
+var proxyTypes = []string{
+	"tor-auto",
+	"socks5",
+}
+
+var proxyTypesNames = map[string]i18n.T{
+	"tor-auto": i18n.T("Automatic Tor"),
+	"socks5":   i18n.T("SOCKS5"),
+}
+
+// findProxyTypeFor returns the index of the proxy type given
+func findProxyTypeFor(s string) int {
+	for ix, px := range proxyTypes {
+		if px == s {
+			return ix
+		}
+	}
+
+	return -1
+}
+
+// getProxyTypeNames will yield all i18n proxy names to the function
+func getProxyTypeNames(f func(string)) {
+	for _, px := range proxyTypes {
+		l := i18n.Local(string(proxyTypesNames[px]))
```

Comment:
So, basically the idea is to use this marker `T` instead of to use/create function `Local`?
If so, you don't need this `i18n.Local` :cat: